spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/reader/ValueRowReader.scala

```diff
@@ -14,7 +14,10 @@ class ValueRowReader[T: TypeConverter](columnRef: ColumnRef) extends RowReader[T
     * @param row row fetched from Cassandra
     * @param rowMetaData: column names available in the `row` */
   override def read(row: Row, rowMetaData: CassandraRowMetadata): T =
-    converter.convert(GettableData.get(row, columnRef.cqlValueName))
+    converter.convert(GettableData.get(
+      row,
+      columnRef.cqlValueName,
+      rowMetaData.codecs(columnRef.cqlValueName)))
```

Comment:
I think what we really want to do may be to set all the row getting machinery once in all RowReaders. So each RowReader gets to have it's own cached list of what columns will be read and in what order ... I do think that's a bit out the scope of this ticket which is just to reduce the massive amount of garbage being made by the full codec lookup